### DISCUSSION TITLE
Provide router object with reportWebVitals

### DISCUSSION THREAD

**Discussion by jasonwilliams** [vercel/next.js#18614] [Ideas]
# Feature request
reportWebVitals is a nice feature, but one thing we’re unable to do is link the stats with a page. We don’t use google analytics which handles his automatically, we usually send the pathname with our stats so we can see which page types perform best.

Right now this is hard to do, we can’t use the `useRouter` hook because reportWebVitals isn’t a component. The only other option is to grab it from `document.next.router` but that feels a bit hacky, should we be using that object?

Having context would also be useful, but if this is not possible I would be happy with just router for now.

**Comment by timneutkens** [comment]
You can currently access it through `import Router from 'next/router'` but we'll likely want to provide the relevant information in the reportWebVitals method like you're suggesting 👍

**Reply by weyert** [reply]
You can access`window.location.href`?

**Reply by jasonwilliams** [reply]
@weyert thats the actual path, I was after the pathName pattern (`/something/[id]`)
